src/plugins/saved_objects/public/finder/saved_object_finder.tsx

```diff
@@ -178,6 +191,25 @@ class SavedObjectFinderUi extends React.Component<
       })
     );
 
+    // Filter search objects whose language are not supported by the current Application
+    savedObjects = savedObjects.filter((obj) => {
+      if (obj.type === 'search') {
+        if (obj.attributes?.kibanaSavedObjectMeta) {
+          const sourceObject = JSON.parse(obj.attributes?.kibanaSavedObjectMeta?.searchSourceJSON);
+          const languageId = sourceObject.query.language;
+
+          const languageProperties = this.languageService.getLanguage(languageId);
+
+          if (languageProperties?.supportedAppNames.includes(currentAppId)) {
```

Comment:
can be similified to

```
return this.languageService.getLanguage(languageId)?.supportedAppNames.includes(currentAppId)) || true
```

Reply:
That makes sense. I have updated the change to reflect this.